src/main/scala/tech/cryptonomic/conseil/tezos/TezosRemotes.scala

```diff
@@ -0,0 +1,202 @@
+package tech.cryptonomic.conseil.tezos
+
+import tech.cryptonomic.conseil.config.{HttpStreamingConfiguration, NetworkCallsConfiguration}
+import tech.cryptonomic.conseil.config.Platforms.TezosConfiguration
+import tech.cryptonomic.conseil.generic.chain.RemoteRpc
+import tech.cryptonomic.conseil.util.JsonUtil.JsonString
+
+
+object TezosRemoteInstances {
+
+  object Akka {
+    import com.typesafe.scalalogging.Logger
+    import cats.Monoid
+    import scala.concurrent.Future
+    import akka.actor.ActorSystem
+    import akka.stream.ActorMaterializer
+    import akka.stream.scaladsl.Source
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model._
+
+    case class RemoteContext(
+      tezosConfig: TezosConfiguration,
+      requestConfig: NetworkCallsConfiguration,
+      streamingConfig: HttpStreamingConfiguration
+    )(implicit val system: ActorSystem) {
+
+      object ShutdownComplete extends ShutdownComplete
+      private val rejectingCalls = new java.util.concurrent.atomic.AtomicBoolean(false)
+
+      def translateCommandToUrl(command: String): String = {
+        import tezosConfig.nodeConfig._
+        s"$protocol://$hostname:$port/${pathPrefix}chains/main/$command"
+      }
+
+      def shutdown(): Future[ShutdownComplete] = {
+        rejectingCalls.compareAndSet(false, true)
+        Http(system).shutdownAllConnectionPools().map(_ => this.ShutdownComplete)(system.dispatcher)
+      }
+
+      def withRejectionControl[T, Container[_]](call: => Container[T])(implicit mono: Monoid[Container[T]]): Container[T] =
```

Comment:
Could you explain what this does and how it works? Why does it need a monoid to the extent that we are creating 'unnatural' monoids on futures?

Reply:
Sure, the weirdness can be explained like this:
- the original code provided a default failed reply for rejection on Future, namely Future.failed(...)
- to define a Rejection result for more generic "effects", we need a failed case default instance for `Container[T]`
- to get a "default" as a type class, Cats offers a specific type to provide such "empty" instance only in a secondary project called alleycats, as in the core project, empty is only available for a Monoid (as the neutral element for the monoid operation "append"). I didn't want to add an additional dependency only for this feature
- after thinking about it, this is not even the correct solution, because when rejecting the call it would be better to mark the response as a "failed" one, thus an `ApplicativeError` or `MonadError` is a better matched choice for this case. I'm in favour of changing the requirement, so that a rejection calls "raiseError"